lib/internal/http2.js

```diff
@@ -1238,12 +1215,17 @@ function sessionOnHeaderComplete(stream, flags, headers, category) {
 }
 
 function connectionListener(socket) {
+  // Turn off the Nagle's for this socket...
+  // highly recommended for http/2 implementations
+  // TODO(jasnell): May want to make this a configuration option?
+  socket.setNoDelay();
```

Comment:
Avoid doing this would cause problems for tiny packets, which are very common in h2.
The standard within node is for Nagle's algorithm to be on. Because disabling Nagle's is _the way to go_ for HTTP/2, we disable it. I don't think we need a config option. An user can always listen for the socket event and call `socket.setNoDelay(false)`.

IMHO providing a configuration option with an "inverse" default compared to the rest of node will be counterintuitive.